decidim-budgets/app/controllers/decidim/budgets/orders_controller.rb

```diff
@@ -39,7 +39,7 @@ def destroy
       private
 
       def budget
-        @budget ||= Budget.find_by(id: params[:budget_id])
+        @budget ||= Budget.where(component: current_component).find_by(id: params[:budget_id])
```

Comment:
I think we could move the component condition inside the `find_by`:
```suggestion
        @budget ||= Budget.find_by(id: params[:budget_id], component: current_component)
```